src/python/tests/core/bot/minimizer/js_minimizer_test.py

```diff
@@ -0,0 +1,166 @@
+    self._minimizer.minimize(data)
+
+    self.assertIn("try{} catch(e){ }", self._tests_to_queue)
+
+  def test_handle_if_else(self):
+    """Make sure the minimizer runs all of the other hypothesis cleanly on
+      if else"""
+    data = "if(boolean) {crash} else { do_something_else }"
+
+    self._minimizer.minimize(data)
+
+    self.assertIn("if(boolean) {}", self._tests_to_queue)
+    self.assertIn("if(boolean) {} else { do_something_else }",
+                  self._tests_to_queue)
+    self.assertIn("if(boolean) {crash} else {}", self._tests_to_queue)
+
+  def test_remove_function_call(self):
+    """Test that it removes functions calls effectively"""
+    data = "function name(param1, param2){ stuff inside function }"
+
+    self._minimizer.minimize(data)
+
+    self.assertIn("function name(param1, param2){}", self._tests_to_queue)
+
+  def test_handle_bracket_with_new_line(self):
+    """Test for try/catch with extra whitespace"""
+    data = """try{\n\tcrash()\n}\ncatch(e){\n\n}"""
+
+    self._minimizer.minimize(data)
+
+    self.assertIn("try{}", self._tests_to_queue)
+    self.assertIn("try{}\ncatch(e){\n\n}", self._tests_to_queue)
+
+  def test_handle_function_with_new_lines(self):
+    """Test for functions with extra whitespace"""
+    data = "function name(param1,\n\t\tparam2){\n\tstuff inside function\n}"
+
+    self._minimizer.minimize(data)
+
+    self.assertIn("function name(param1,\n\t\tparam2){}", self._tests_to_queue)
+
+  def test_remove_outer_paren(self):
+    """e.g.: assertTrue(crash()); -> crash()"""
+    data = "assertTrue(crash());"
+
+    self._minimizer.minimize(data)
+
+    self.assertIn("assertTrue()", self._tests_to_queue)
+
+  def test_remove_inside_paren(self):
+    """Hypothesis is Remove everything between the parentheses.
+      e.g.: crash(junk, more_junk) -> crash()"""
+    data = "crash(junk, more_junk)"
+
+    self._minimizer.minimize(data)
+
+    self.assertIn("junk, more_junk", self._tests_to_queue)
+
+  def test_remove_paren_to_start_of_line(self):
+    """e.g.: leftover_junk = (function() {
+             });"""
+    data = "leftover_junk = (function(){\n})"
+
+    self._minimizer.minimize(data)
+
+    self.assertIn("leftover_junk = (function(){\n})", self._tests_to_queue)
+
+  def test_remove_paren_with_attached_brackets(self):
+    """e.g.: (function(global) { })(this);"""
+    data = "(function(global) { })(this)"
+
+    self._minimizer.minimize(data)
+
+    self.assertIn("(function(global) { })(this)", self._tests_to_queue)
+
+  def test_remove_left_of_comma(self):
+    """e.g.: f(whatever, crash()) -> f(crash())"""
+    data = "f(whatever, crash())"
+
+    self._minimizer.minimize(data)
+
+    self.assertIn("whatever,", self._tests_to_queue)
+
+  def test_remove_right_of_comma(self):
+    """e.g.: f(crash(),whatever) -> f(crash())"""
+    data = "f(crash(), whatever)"
+
+    self._minimizer.minimize(data)
+
+    self.assertIn(", whatever", self._tests_to_queue)
```

Comment:
What is _tests_to_queue, all these tests are very confusing. these should test what is going on in the iterations, but also validate the final minimized result.

Reply:
_tests_to_queue is the name of the queue in minimizer that keeps track of which hypotheses need to be tested. I agree that the name is not clear outside the context of the minimizer. 

The tests are all validating that the minimizer is attempting to remove the correct tokens at some point in the run. We mocked prepare_test to just add the hypotheses to a list, and skip actually processing any of the tests. This way the tests can validate that the correct tokens were removed and the tests run much faster than running the minimizer in full.